### PR TITLE
Everywhere: Mark most HashMap copies explicitly, avoid some unnecessary copies

### PR DESCRIPTION
**Pull request by BenWiederhake**
While working on #18759, I noticed that we permitted implicit copy-construct and copy-assign for `HashMap`. Since this type can become reasonably large, it seems like a good idea to avoid implicit copies of these, as they are more likely to encounter `ENOMEM` than, say, a 16-byte struct.

This PR looks at most implicit HashMap copies, and either:
- avoids the copy entirely, e.g. turning it into a move or a reference, or
- marks the copy explicitly as copies (using `HashMap::clone`), or
- leaves it as an implicit copy, if it is buried in another implicit copy (e.g. the implicit copy-constructor of a `struct Foo {HashMap<Bar, Baz> entries;};`). Examples are `TLS::Certificate`, or `Web::HTML::ImportMap`.

The effect should be:
- marginal memory saving, if any; and
- easier error handling if the copying fails; and
- ideally in the future someone can finish the hunt and turn it into `AK_MAKE_NONCOPYABLE(HashMap);`

In case you're curious, you can do a checkout and see how many callsites remain:
```
    HashMap(HashMap const&) = delete;
    HashMap(HashMap&& other) noexcept = default;
    HashMap& operator=(HashMap const& other) = delete;
    HashMap& operator=(HashMap&& other) noexcept = default;
```

### CODE REVIEW

**Comment by BenWiederhake** [comment]
Changes since last push:
- Add a commit that explicitly permits default {copy,move}-{construct,assign} implementations.
- Add a FIXME for the OOM-problem to the copy-{construct,assign} calls.

**Comment by BenWiederhake** [comment]
Changes since last push:
- Rebased, inspired by toolchain update
- Added template arguments to `HashMap::clone()` so that cloning into a slightly different type of HashMap becomes possible
- Fixed the commit subject for that commit, since it wasn't just about "honoring" some args.

File: AK/HashMap.h [hunk 1]
```diff
@@ -40,6 +40,11 @@ class HashMap {
             set(item.key, item.value);
     }
 
+    HashMap(HashMap const&) = default; // FIXME: Not OOM-safe! Use clone() instead.
+    HashMap(HashMap&& other) noexcept = default;
+    HashMap& operator=(HashMap const& other) = default; // FIXME: Not OOM-safe! Use clone() instead.
+    HashMap& operator=(HashMap&& other) noexcept = default;
```
Comment (awesomekling) [inline AK/HashMap.h:46]: How do these comments "reveal" that we copy-construct and copy-assign HashMaps very often?

Comment (BenWiederhake) [reply AK/HashMap.h:46]: You're right, "reveal" is a bit too strong. How about "underlines"?

**Comment by BenWiederhake** [comment]
Changes since last push:
- Rebased for convenience
- Changed the last commit body to be less of an exaggeration.

**Review by kleinesfilmroellchen — APPROVED** [review]

File: Meta/Lagom/Tools/CodeGenerators/LibUnicode/GenerateUnicodeData.cpp [hunk 2]
```diff
@@ -1443,7 +1443,7 @@ static void normalize_script_extensions(PropList& script_extensions, PropList co
     // abbreviation. Further, the extensions may or may not include the base Script list. Normalize
     // the extensions here to be keyed by the full Script name and always include the base list.
     auto extensions = move(script_extensions);
-    script_extensions = script_list;
+    script_extensions = MUST(script_list.clone());
```
Comment (GMTA) [inline Meta/Lagom/Tools/CodeGenerators/LibUnicode/GenerateUnicodeData.cpp:1446]: `normalize_script_extensions` is only invoked in one function that returns an `ErrorOr`; maybe you can make this a `TRY` instead and return `ErrorOr<void>`?

Comment (BenWiederhake) [reply Meta/Lagom/Tools/CodeGenerators/LibUnicode/GenerateUnicodeData.cpp]: Absolutely! Weird, I could have sworn I checked that before writing "MUST".

**Comment by BenWiederhake** [comment]
Changes since last push:
- Rebased for convenience
- Changed an unnecessary MUST to a TRY (Thanks @GMTA!)

**Review by GMTA — APPROVED** [review]

**Pull request merged**